### PR TITLE
chore: various fixes for Python benchmarks

### PR DESCRIPTION
**Pull request by wjones127**
Doing a bit of maintenance on Python benchmarks:

1. Added the `--features datagen` to the instructions, since that is now required for some benchmarks.
2. Registered a pytest mark for `cuda` so we stop getting a warning from pytest.
3. Updated comparison instructions to use main branch, since that will give more comparable results given we don't compile with the same arguments when we release.
4. Add instructions on how to use `trace_to_chrome` with the benchmarks.
5. Reduce the size of the KMeans test to something that runs in a reasonable amount of time. On my machine it takes over a minute too finish, which is too slow for iteration.
6. Fix tracing so it filters out other packages messages.

### CODE REVIEW

File: python/python/benchmarks/test_kmeans.py [hunk 1]
```diff
@@ -16,12 +16,11 @@
 import pytest
 from lance.util import KMeans
 
-CLUSTERS = 1024
+CLUSTERS = 32
```
Comment (eddyxu) [inline python/python/benchmarks/test_kmeans.py:19]: kmean is `O(n^2)` to # of clusters. Would this number too small

Comment (wjones127) [reply python/python/benchmarks/test_kmeans.py:19]: > Would this number too small

What do you mean? Too small for what? Does this trigger different code paths if the values is lower?

Right now this benchmark takes far too long. I'm trying to keep these benchmarks so they can all run in just a few minutes. Most of these should just take a second or two to run. We can get plenty of samples for tracing and flamegraphs in that time. This single benchmark takes 10+ minutes on my laptop, which is annoying to wait for. We can always run one-off large scale benchmarks when we want to wait longer for the result.

Comment (eddyxu) [reply python/python/benchmarks/test_kmeans.py:19]: I see. it is fair.

Comment (wjones127) [reply python/python/benchmarks/test_kmeans.py]: I've restored this test as it was, but put it behind a `"slow"` mark so it's easy to filter out.

**Comment by wjones127** [comment]
@eddyxu I think this is ready to go.

**Review by westonpace — APPROVED** [review]
This looks like a helpful cleanup to me

File: python/python/benchmarks/test_index.py [hunk 2]
```diff
@@ -53,6 +53,11 @@ def test_create_ivf_pq(test_dataset, benchmark):
 @pytest.mark.benchmark(group="create_index")
 @pytest.mark.cuda
 def test_create_ivf_pq_cuda(test_dataset, benchmark):
+    import torch
+
+    if not torch.cuda.is_available():
+        pytest.skip("No CUDA device available")
```
[the remark this answers is not in the document]
Comment (wjones127) [reply python/python/benchmarks/test_index.py]: Ah as it turns out, the problem is that the `conftest.py` configuration that did this was in the tests directory, and not in the benchmarks directory. I pulled out the common configuration into a `conftest.py` in the base `python` directory so now it applies to both.

[the remark this answers is not in the document]
Comment (wjones127) [reply python/python/benchmarks/test_index.py]: Plot twist: adding `conftest.py` in the `python` directory will cause that directory to be added to the Python path. That means the `python/lance` directory will be picked up as a module and preferred over the installed `pylance` module for unit tests, causing them to fail in our Linux testing setup. Therefore, I'd had to resort to duplicating the logic across the two `conftest.py` files.

**Pull request merged**